test/production/edge-dynamic-code-eval/index.test.ts

```diff
@@ -0,0 +1,55 @@
+import { nextTestSetup } from 'e2e-utils'
+
+describe('Dynamic Code Evaluation (DCE)', () => {
+  const { next } = nextTestSetup({
+    skipStart: true,
+    dependencies: { 'function-bind': 'latest' },
+    files: __dirname,
+  })
+
+  // This test is basically for https://github.com/vercel/next.js/discussions/51910
+  // to make sure that some libs that we know are using `eval` but don't break
+  // because it will never run into that condition, but still can't to be DCE'd.
+  it('should not fail when "function-bind" package is used', async () => {
+    await next.start()
+
+    expect(next.cliOutput).not.toContain(
+      `Dynamic Code Evaluation (e. g. 'eval', 'new Function', 'WebAssembly.compile') not allowed in Edge Runtime`
+    )
+
+    await next.stop()
+  })
+
+  it("should show the user's import trace", async () => {
+    await next.patchFile(
+      'middleware.js',
+      `
+      import { foo } from './lib/foo'
+      export function middleware() {
+        foo()
+      }`
+    )
+    const output = await next.build()
+    expect(output.exitCode).toBe(1)
+    expect(output.cliOutput.split('\n').slice(3).join('\n'))
```

Comment:
We can use `toContain` to check different parts are included:

toContain(Dynamic Code Evaluation ...) // the text
toContain(Import trace for requested module: ...) // the module trace

Inline snapshot might easily change if we changed sth, since we don't have strong requirement to match the format we can just check if the cirtical content are included